pkg/vault/test/foundry/LiquidityApproximation.t.sol

```diff
@@ -199,11 +200,21 @@ contract LiquidityApproximationTest is BaseVaultTest {
 
         assertEq(dai.balanceOf(alice), dai.balanceOf(bob), "Bob and Alice DAI balances are not equal");
 
+        if (usdc.balanceOf(alice) <= defaultBalance) {
```

Comment:
Can it actually be less (due to rounding, I guess)? If not, it would be self-documenting to move the `aliceAmountOut` calculation up above this line and just check for `aliceAmountOut == 0` here. If it can actually be less though, you'd have to do it this way, as otherwise you'd get an underflow calculating the difference. Here and one place below.

Reply:
It can be less for very small amounts (this PR just paves the way to fuzz small ranges harder). Might be related to this: https://github.com/balancer/balancer-v3-monorepo/pull/946.

Indeed, I've implemented the test this way to avoid underflows downstream.